process.go

```diff
@@ -88,37 +295,152 @@ func (p Process) Validate() error {
 			}
 		}
 	}
+	return nil
+}
+
+// ProcessPort is network port information for a workload process.
+type ProcessPort struct {
+	// External is the port on the host.
+	External int
+	// Internal is the port on the process.
+	Internal int
+	// Endpoint is the unit-relation endpoint matching the external
+	// port, if any.
+	Endpoint string
+}
+
+// Set parses the provided string and sets the appropriate fields.
+func (p *ProcessPort) Set(raw string) error {
+	parts := strings.SplitN(raw, ":", 2)
+	if len(parts) != 2 {
+		return fmt.Errorf("invalid value %q", raw)
+	}
+	if err := p.SetExternal(parts[0]); err != nil {
+		return err
+	}
+	if err := p.SetInternal(parts[1]); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetExternal parses the provided string and sets the appropriate fields.
+func (p *ProcessPort) SetExternal(portStr string) error {
+	p.External = 0
+	p.Endpoint = ""
+	if strings.HasPrefix(portStr, "<") && strings.HasSuffix(portStr, ">") {
+		// The port was specified by a relation endpoint rather than a
+		// port number.
+		p.Endpoint = portStr[1 : len(portStr)-1]
+	} else {
+		// It's just a port number.
+		port, err := strconv.Atoi(portStr)
+		if err != nil {
+			return fmt.Errorf("expected int got %q", portStr)
+		}
+		p.External = port
+	}
+	return nil
+}
+
+// SetInternal parses the provided string and sets the appropriate fields.
+func (p *ProcessPort) SetInternal(portStr string) error {
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		return fmt.Errorf("expected int got %q", portStr)
+	}
+	p.Internal = port
+	return nil
+}
+
+// ProcessVolume is storage volume information for a workload process.
+type ProcessVolume struct {
+	// ExternalMount is the path on the host.
+	ExternalMount string
+	// InternalMount is the path on the process.
+	InternalMount string
+	// Mode is the "ro" OR "rw"
+	Mode string
```

Comment:
I don't think so.  Since Go doesn't check the values, a typedef doesn't help much, particularly since it's a string.  In fact it kind of makes the code messier and harder to follow.  We have all the validation we need for the two values so I think it's fine as-is.  Given that we rely pretty heavily on serialization and parsing for the metadata, let's just keep this as strings.